src/libcore/iter.rs

```diff
@@ -2423,7 +2425,11 @@ macro_rules! step_impl {
             #[allow(trivial_numeric_casts)]
             fn steps_between(start: &$t, end: &$t, by: &$t) -> Option<usize> {
                 if *start <= *end {
-                    Some(((*end - *start) / *by) as usize)
+                    // Note: We assume $t <= usize/isize here
+                    Some(
+                        ((*end as isize).wrapping_sub(*start as isize) as usize
```

Comment:
You may want to add a few test cases that use integer types larger than `i8` and `i16` as well because they'll always fit into an `isize` right now.